### PR TITLE
Feed and notifications

### PR DESCRIPTION
**Pull request by f1nwe**

### CODE REVIEW

File: app/controllers/concerns/feed.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Feed
+  def respond_for_feed
+    if request.xhr?
+      render_xhr_response
+    else
+      render 'shared/feed'
+    end
+  end
+
+  def render_xhr_response
```
Comment (denys281) [inline app/controllers/concerns/feed.rb]: Change name for more specific.

File: app/controllers/concerns/feed.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    if @events.any?
```
Comment (denys281) [inline app/controllers/concerns/feed.rb]: Don't use instance variable here, pass it to method.

File: app/models/user.rb [hunk 3]
```diff
@@ -35,4 +37,24 @@ def self.user_params(auth)
...
+    notifications.unread.any?
+  end
+
+  def new_notifications_for?(event)
+    notifications.find_by(event_id: event.id, readed: false)
+  end
+
+  def read_notifications_for(events)
+    events.each do |e|
+      notification = notifications.find_by(event_id: e.id, readed: false)
```
Comment (denys281) [inline app/models/user.rb]: User `new_notifications_for?(event)` here

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
+    current_user.read_notifications_for(@events) if @events.present?
+  end
+
   private
 
   def find_user
     @user = User.find(params[:id])
   end
+
+  def load_events
```
Comment (denys281) [inline app/controllers/users_controller.rb]: see skype :)

File: app/assets/javascripts/feed.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    start_scroll_listening();
+    // end of events autoload
+
+    // Helpers
+
+    function end_of_page() {
+        return $(window).scrollTop() + $(window).height() > $(document).height() - 100
+    }
+
+    function load_events() {
```
Comment (AndriySolonyna) [inline app/assets/javascripts/feed.js]: `$.ajax({
  url: 'mypage.html',
  success: function(){
    alert('success');
  },
  error: function(){
    alert('failure');
  },
  beforeSend: function(){
    $('#loader-block').removeClass('hidden');
  },
  complete: function() {
     $('#loader-block').addClass('hidden');
  }
});`

**Pull request closed without merging**